### PR TITLE
Add math_utils.py

### PR DESCRIPTION
**Pull request by bonicim**
What is the change?
* Add math_utils.py to volttron.utils
* Add codacy config file
* Add prospector config file
* Add pydocstyle hook to pre-commit file

Why is it needed?
* math_utils.py is used by agents to do some common math routines. This PR provides the math support required by [PlatformDriverAgent](https://github.com/VOLTTRON/volttron-platform-driver).
* We keep failing code checks on [Codacy](https://app.codacy.com/gh/VOLTTRON/volttron-core/dashboard) on certain mutually exclusive rules. Specifically, Codacy will fail our PR's on two [doctring rules, D212 and D213, which are mutually exclusive](https://github.com/KratosMultiphysics/Kratos/issues/8816). The current behavior of Codacy is creating false negatives (i.e. The PR does not meet code standards) on our PR's. We need to configure Codacy to ignore these mutually exclusive rules. The documentation used to configure Codacy is found [here](https://docs.codacy.com/repositories-configure/codacy-configuration-file/). Some examples of codacy configuration files can be found [here](https://earth.bsc.es/gitlab/es/earthdiagnostics/-/commit/fab7addcc939e35766b4ca79ae7b3b520b8f718e). 
* Prospector is the tool that is causing false negatives. The prospector config file is used to remove the mutually exclusive rules.  
* We should catch docstring errors locally and as part of our CI. By including [pydocstyle](https://github.com/PyCQA/pydocstyle), we can catch the errors before they hit Codacy.

### CODE REVIEW

**Comment by schandrika** [comment]
@bonicim could you check the standard deviation methods. Shouldn't the last step be a square root (math.sqrt(x)) and not *0.5.

**Comment by bonicim** [comment]
> @bonicim could you check the standard deviation methods. Shouldn't the last step be a square root (math.sqrt(x)) and not *0.5.

@schandrika 
There's another asterisk, [which is used to get the square root](https://www.thecrazyprogrammer.com/2018/05/how-to-find-square-root-in-python.html),  in that line of code. 

```
return pvar**0.5
```

By using another asterisk, the function gets the square root by using exponents as seen in the example below:

Screen Shot 2022-05-09 at 3 06 52 PM


But I like your suggestion on using the math library. It is `declarativish` and easier to read. I will use the math library to get the square root instead of the cryptic double asterisks.

**Comment by craig8** [comment]
@bonicim please add 

from volttron.utils import math_utils as math 

to the __init__.py in volttron.utils

**Comment by craig8** [comment]
As far as the other imports, yeah they should be changed, but not in this pull request....we will move the __all__ message from them up into this one....

File: volttron/utils/__init__.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 # BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
 # under Contract DE-AC05-76RL01830
 # }}}
+"""Some important modules are imported here to allow for implicit imports of such modules."""
```
Comment (craig8) [inline volttron/utils/__init__.py:38]: ? aren't all modules that are used important.

This is the doc string so it should be something descriptive of what the4 utils package has in it.

"""
The volttron.utils package contains generic utilities for handling json, storing configurations
math libraries...and more
"""

**Comment by craig8** [comment]
@bonicim can we add a few tests around the functions that we are running in the math utils.  In the spirit of eventual code cov I think that would be good.

**Pull request merged**